client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c

```diff
@@ -779,10 +779,13 @@ kaa_error_t kaa_tcp_channel_process_event(kaa_transport_channel_interface_t *sel
                         error_code = kaa_tcp_channel_write_pending_services(tcp_channel
                                 , tcp_channel->pending_request_services
                                 , tcp_channel->pending_request_service_count);
-                    } else {
+                    } else if (tcp_channel->channel_state != KAA_TCP_CHANNEL_AUTHORIZING){
```

Comment:
Please, do `else if (tcp_channel_state == KAA_TCP_CHANNEL_AUTHORIZING)` and swap branches. It's easier to understand this way (more uniform).